lib/iiif/v3/presentation/nav_place.rb

```diff
@@ -0,0 +1,98 @@
+require 'geo/coord'
+
+module IIIF
+  module V3
+    module Presentation
+      class NavPlace < IIIF::V3::AbstractResource
+        Rect = Struct.new(:coord1, :coord2)
+
+        def initialize(coordinate_texts:, base_uri:)
+          @coordinate_texts = coordinate_texts
+          @base_uri = base_uri
+        end
+
+        def build
+          return if coordinates.nil? || coordinates.empty?
+
+          {
+            id: "#{base_uri}/feature-collection/1",
+            type: 'FeatureCollection',
+            features:
+          }
+        end
+
+        private
+
+        attr_reader :coordinate_texts, :base_uri
+
+        def coordinates
+          @coordinates ||= coordinate_texts.map do |coordinate_text|
+            coordinate_parts = coordinate_text.split(%r{ ?--|/})
+            case coordinate_parts.length
+            when 2
+              coord_for(coordinate_parts[0], coordinate_parts[1])
+            when 4
+              rect_for(coordinate_parts)
+            end
+          end.compact
+        end
+
+        COORD_REGEX = /(?<hemisphere>[NSEW]) (?<degrees>\d+)[°⁰*] ?(?<minutes>\d+)?[ʹ']? ?(?<seconds>\d+)?[ʺ"]?/
+
+        def coord_for(long_str, lat_str)
+          long_matcher = long_str.match(COORD_REGEX)
+          lat_matcher = lat_str.match(COORD_REGEX)
+          return unless long_matcher && lat_matcher
+
+          Geo::Coord.new(latd: lat_matcher[:degrees], latm: lat_matcher[:minutes], lats: lat_matcher[:seconds], lath: lat_matcher[:hemisphere],
+                        lngd: long_matcher[:degrees], lngm: long_matcher[:minutes], lngs: long_matcher[:seconds], lngh: long_matcher[:hemisphere])
+        end
+
+        def rect_for(coordinate_parts)
+          coord1 = coord_for(coordinate_parts[0], coordinate_parts[2])
+          coord2 = coord_for(coordinate_parts[1], coordinate_parts[3])
+          return if coord1.nil? || coord2.nil?
+
+          Rect.new(coord1, coord2)
+        end
+
+        def features
+          coordinates.map.with_index do |coordinate, index|
```

Comment:
If you pass a starting value to `with_index`, then you won't need to add to it on every loop.